server/api/users/users.test.ts

```diff
@@ -37,19 +37,66 @@ describe('auth router', () => {
   });
 
   describe('GET /', () => {
-    it('expects empty array if no users exist', async () => {
+    it('expects empty array if no users exist on the homepage', async () => {
       const res = await server.exec.get('/api/users/');
-      expect(res.body).toStrictEqual([]);
+      expect(res.body.users).toStrictEqual([]);
     });
 
-    it('returns all users', async () => {
-      const userOne = await createUser();
-      const userTwo = await createUser();
+    it('expect invalid page request to return invalid page', async () => {
+      const res = await server.exec.get('/api/users/?page=-1');
+      expect(res.status).toBe(400);
+      expect(res.body.error).toEqual('Invalid page number');
+    });
+
+    it('expect page request beyond page limit to be out of range', async () => {
+      const res = await server.exec.get('/api/users/?page=100');
+      expect(res.status).toBe(400);
+      expect(res.body.error).toEqual('Page out of range');
+    });
+
+    it('returns first 20 users when it hits the homepage', async () => {
+
+      const users: User[] = [];
+
+      for(let i = 0; i < 25; i++){
+        const user = await createUser();
+        users.push(user);
+      }
+
+      const limit = 20;
+      const count = await User.count();
+      const totalPages = Math.ceil(count / limit);
 
       const res = await server.exec.get('/api/users/');
       expect(res.status).toBe(200);
-      expect(res.body[0]).toEqual(getExpectedUserObject(userOne));
-      expect(res.body[1]).toEqual(getExpectedUserObject(userTwo));
+      expect(res.body.users.length).toEqual(limit);
+      expect(res.body.users[0]).toEqual(getExpectedUserObject(users[0]));
+      expect(res.body.users[19]).toEqual(getExpectedUserObject(users[19]));
```

Comment:
i dont think we actually need to validate that the first and last users match, only we are receiving 20 users.  20 distinct users if you want to go the extra mile but i don't think it's necessary